homeassistant/components/switch/amcrest.py

```diff
@@ -0,0 +1,92 @@
+"""
+Support for toggling Amcrest IP camera settings.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.amcrest/
+"""
+import asyncio
+import logging
+
+from homeassistant.components.amcrest import DATA_AMCREST, SWITCHES
+from homeassistant.const import (
+    CONF_NAME, CONF_SWITCHES, STATE_OFF, STATE_ON)
+from homeassistant.helpers.entity import ToggleEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['amcrest']
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the IP Amcrest camera switch platform."""
+    if discovery_info is None:
+        return
+
+    name = discovery_info[CONF_NAME]
+    switches = discovery_info[CONF_SWITCHES]
+    camera = hass.data[DATA_AMCREST][name].device
+
+    all_switches = []
+
+    for setting in switches:
+        all_switches.append(AmcrestSwitch(setting, camera))
+
+    async_add_devices(all_switches, True)
+
+
+class AmcrestSwitch(ToggleEntity):
+    """Representation of an Amcrest IP camera switch."""
+
+    def __init__(self, setting, camera):
+        """Initialize the Amcrest switch."""
+        self._setting = setting
+        self._camera = camera
+        self._name = SWITCHES[setting][0]
+        self._icon = SWITCHES[setting][1]
+        self._state = None
+
+    @property
+    def name(self):
+        """Return the name of the switch if any."""
+        return self._name
+
+    @property
+    def state(self):
```

Comment:
This should not be overwrittten for switch entities. It's handled by the `ToggleEntity` base class.